kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/subscription/SpecificationObserverWrapper.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa.subscription
+
+import android.util.Log
+import org.eclipse.kuksa.PropertyObserver
+import org.eclipse.kuksa.VssSpecificationObserver
+import org.eclipse.kuksa.extension.TAG
+import org.eclipse.kuksa.extension.copy
+import org.eclipse.kuksa.proto.v1.Types
+import org.eclipse.kuksa.vsscore.model.VssSpecification
+
+internal class SpecificationObserverWrapper<T : VssSpecification>(
+    specification: T,
+    vssPaths: List<String>,
+    private val observer: VssSpecificationObserver<T>,
+) : PropertyObserver {
```

Comment:
So it is a SpecificationPropertyObserver? Because it wraps through all properties of the specification? Naming :)

Reply:
Fine with me! I wanted to emphasize the fact that it simply wraps the SpecificationObserver, but yeah the naming is weird, even more so since it's not clear that it's a PropertyObserver